dotnet/src/SemanticKernel.Abstractions/Kernel.cs

```diff
@@ -105,6 +114,8 @@ public sealed class Kernel
             PromptRendering = this.PromptRendering,
             PromptRendered = this.PromptRendered,
             _data = this._data is { Count: > 0 } ? new Dictionary<string, object?>(this._data) : null,
+            _functionFilters = this._functionFilters is { Count: > 0 } ? new List<IFunctionFilter>(this._functionFilters) : null,
+            _promptFilters = this._promptFilters is { Count: > 0 } ? new List<IPromptFilter>(this._promptFilters) : null,
```

Comment:
Both `_functionFilters` and `_promptFilters` fields are initialized in the class constructor - line 110, which is executed before the execution control reaches these two lines - 117 and 118. Do we still need to overwrite them here?

Reply:
Good catch, I removed these lines and also added check in unit-tests that filters are cloned correctly.